src/test/osd/TestPGLog.cc

```diff
@@ -2064,6 +2066,269 @@ TEST_F(PGLogTest, ErrorNotIndexedByObject) {
   EXPECT_EQ(del.reqid, entry->reqid);
 }
 
+class PGLogMergeDupsTest : public ::testing::Test, protected PGLog {
```

Comment:
👍 on unit tests for trimming.

There also aren't any explicit tests that it actually reports duplicate ops. :o Is that somehow covered by existing unit tests?